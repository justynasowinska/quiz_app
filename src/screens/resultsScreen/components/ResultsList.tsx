import * as React from 'react';
import { FlatList, View, StyleSheet, ListRenderItemInfo } from 'react-native';
import { ResultType, ResultsListItem } from './ResultsListItem';

export type ResultsType = ReadonlyArray<ResultType>;

interface PropsType {
    results: ResultsType;
}

export const ResultsList = (props: PropsType) => {
    const { results } = props;

    const renderListItem = ({ item }: ListRenderItemInfo<ResultType>) => {
        return (
            <ResultsListItem
                resultsItem={item}
            />
        );
    };

    const keyExtractor = (item: ResultType) => `${item.questionId}`;

    return (
        <View style={styles.container}>
            <FlatList
                data={results}
                renderItem={renderListItem}
                keyExtractor={keyExtractor}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        marginTop: 20,
        marginBottom: 20
    }
});