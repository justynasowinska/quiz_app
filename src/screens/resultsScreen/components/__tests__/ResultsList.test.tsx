import * as React from 'react';
import { shallow } from 'enzyme';

import { ResultsList } from '../ResultsList';

describe('Results Screen ResultsList Component', () => {
    const results = [
        {
            questionId: 0,
            question: 'Unturned originally started as a Roblox game.',
            correct: true,
            correctAnswer: 'True'
        },
        {
            questionId: 1,
            question: 'Unturned originally started as a Roblox game.',
            correct: false,
            correctAnswer: 'False'
        },
        {
            questionId: 3,
            question: 'Unturned originally started as a Roblox game.',
            correct: true,
            correctAnswer: 'False'
        }
    ];

    it('matches to the snapshot', () => {
        const component = shallow(
            <ResultsList
                results={results}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
