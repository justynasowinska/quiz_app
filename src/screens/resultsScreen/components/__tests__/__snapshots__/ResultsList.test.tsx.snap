// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Screen ResultsList Component matches to the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 20,
      "marginTop": 20,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "correct": true,
          "correctAnswer": "True",
          "question": "Unturned originally started as a Roblox game.",
          "questionId": 0,
        },
        Object {
          "correct": false,
          "correctAnswer": "False",
          "question": "Unturned originally started as a Roblox game.",
          "questionId": 1,
        },
        Object {
          "correct": true,
          "correctAnswer": "False",
          "question": "Unturned originally started as a Roblox game.",
          "questionId": 3,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
